i18n/extract.py

```diff
@@ -57,6 +57,30 @@ def add_args(self):
         Adds arguments
         """
         self.parser.description = __doc__
+        self.parser.add_argument(
+            '--merge-js',
+            action='store_true',
+            help='Merge djangojs.po with django.po using msgcat'
+        )
+        self.parser.add_argument(
+            '--combine-partial',
+            action='store_true',
+            help=(
+                'Renames (django-partial.po) and (djangojs-partial.po) to (django.po) and (djangojs.po) respectively. '
+                'This will also remove (django-saved.po) and (djangojs-saved.po) that are used to restore original '
+                '(django.po) and (djangojs.po) after extract is completed. '
+                'This is helpful when this `extract` command output is needed to be used immediately '
+                'with no segment/merge workflow in small repositories.'
+            )
```

Comment:
I still don't think this option is easy to understand or generic enough to become an option in `i18n_tool`.

May I suggest the following combination of both `--combine-partial` and `--text-domain`?

```
self.parser.add_argument(
            '--partial-filename-format',
            default='django{js}-partial.po',
            help=(
                'The file name Python ".format()" template for the partial files. Default is "django{js}-partial.po" which '
                'generates two files: "django-partial.po" and "djangojs-partial.po". Alternative formats such as '
                '"text{js}.po" and "django{js}.po" can be used. '
                'If the "django{js}.po" is used, existing "django.po" and/or "djangojs.po" files are going to be overridden.' 
            )
        )
```

I'm trying to think of a new comer to this repo and they'd read the new option. It should make complete sense to them in a relatively short time.

The current options are focused with implementation details. I'm trying to suggest an option that focuses on the interface i.e. input and output of the tool and hide implementation details.


#### Usage

Credentials theme:
```
i18n_tool extract --partial-filename-format='django{js}.po'
```

XBlocks with merged js domain: 

```
i18n_tool extract --merge-js --partial-filename-format='text{js}.po' 
```

XBlocks with split js domain:

```
i18n_tool extract --partial-filename-format='text{js}.po'
```

Reply:
Thank you @OmarIthawi

The (partial file issue) is that the tool generates `-partial` files and leaves the decision to the developer to replace django(js).po` with the partial file. `edx-platform` needs to keep generating partial files, while other packages don't

For --'text-domain` it's just about the name, do we need the default `django` or the special `text` for XBlocks

So, maybe we just need to rename `--combine-partial` to `--no-partial`. And leave `--text-domain`. I see `--no-partial` can clarify the purpose, which is (don't generate partial files, generate original files)

What do you think?

Credentials theme:

```
i18n_tool extract --no-partial
```

XBlocks with merged js domain:

```
i18n_tool extract --no-partial --merge-js --text-domain
```

XBlocks with split js domain:

```
i18n_tool extract --no-partial --text-domain
```

Reply:
The `--no-partial` seems understandable to me.

If we're going to use `--text-domain` it should be clear that it's going to be deprecated i.e. `--legacy-text-domain` because the `text.po` is an Open edX invention that's not found anywhere in Python `gettext` projects as far as I know. Nevertheless, we're going to deprecate this method of using `text.po` files altogether.

An alternative to the `--legacy-text-domain` is a flat out rename:

```
i18n_tool extract --no-partial --merge-js
rename -f django.po text.po
```

I think this rename isn't worth an option in the `i18n-tools`. Also, pardon me if I've changed my mind multiple times. I'm a bit concerned that we're making this tool somewhat complicated since I'm now one of the few Core Contributors who have maintenance responsibility of this package.

I did a GitHub search for text.po files and almost all results are from XBlocks: https://github.com/search?q=path%3ALC_MESSAGES%2Ftext.po&type=code&p=1

Reply:
The issue is important and it will affect many repositories. Thank you for being concerned about it @OmarIthawi

OK, now we have two options, let me know your last decision 😁 

* Remove the flat rename logic and depend on a deprecating option `--legacy-text-domain` which will raise an error when deprecated. [openedx-tranlations](https://github.com/openedx/openedx-translations) will report that error in its daily `extract_translations` tasks
* Keep using flat rename logic. Removing the logic will be handled by developers when `text.po` is announced to be deprecated

Reply:
I recommend not adding the `--text-domain` option at all and use an additional command in each XBlock:

```
make extact_translations:
    i18n_tool extract --no-partial --merge-js
    rename -f django.po text.po
```

After the `text.po` is completely deprecated, the commands should be updated in each individual XBlock.

This way the `i18n-tools` stays independent of the deprecation of the `text.po` altogether.

Reply:
all done @OmarIthawi , but I used `mv $(EXTRACT_DIR)/django.po $(EXTRACT_DIR)/text.po` instead of `rename` to avoid installing `rename` package for MacOS guys